scheduler/scheduler-examples/src/main/java/org/ow2/proactive/scheduler/examples/SFTPConnector.java

```diff
@@ -0,0 +1,400 @@
+
+    private static final int DEFAULT_SFTP_PORT = 22;
+
+    private static final String GET = "GET";
+
+    private static final String PUT = "PUT";
+
+    private static final String STRICT_HOST_KEY_CHEKING = "StrictHostKeyChecking";
+
+    private static final String CURRENT_FOLDER = ".";
+
+    private static final String PARENT_FOLDER = "..";
+
+    private String sftpHostname = DEFAULT_SFTP_HOSTNAME;
+
+    private int sftpPort = DEFAULT_SFTP_PORT;
+
+    private boolean sftpExtractArchive;
+
+    private String sftpLocalAbsolutePath;
+
+    private String sftpRemoteRelativePath;
+
+    private String sftpMode;
+
+    private String sftpUsername;
+
+    private String sftpUrlKey;
+
+    private String sftpPassword;
+
+    @Override
+    public void init(Map<String, Serializable> args) throws Exception {
+        if (args.containsKey("sftpHostname")) {
+            sftpHostname = args.get("sftpHostname").toString();
+        }
+        if (args.containsKey("sftpPort")) {
+            try {
+                sftpPort = Integer.parseInt(args.get("sftpPort").toString());
+            } catch (NumberFormatException e) {
+                getOut().println("Invalid port number. Using default sftp port=" + DEFAULT_SFTP_PORT);
+            }
+        }
+        if (args.containsKey("sftpMode")) {
+            sftpMode = args.get("sftpMode").toString();
+        }
+        if (args.containsKey("sftpExtractArchive")) {
+            sftpExtractArchive = Boolean.parseBoolean(args.get("sftpExtractArchive").toString());
+        }
+        if (args.containsKey(SFTP_LOCAL_RELATIVE_PATH_ARG) &&
+            !args.get(SFTP_LOCAL_RELATIVE_PATH_ARG).toString().isEmpty()) {
+            sftpLocalAbsolutePath = Paths.get(getLocalSpace(), args.get(SFTP_LOCAL_RELATIVE_PATH_ARG).toString())
+                                         .toString();
+        } else {
+            //we throw an exception tp prevent transferring all the contents of the global space.
+            if (sftpMode.equals(PUT)) {
+                throw new IllegalArgumentException("Please specify a local relative path. Empty value is not allowed.");
+            }
+            //default value is getlocalspace() because it will always be writable and moreover can be used to transfer files to another data space(global, user)
+            sftpLocalAbsolutePath = getLocalSpace();
+        }
+        if (args.containsKey("sftpRemoteRelativePath")) {
+            sftpRemoteRelativePath = args.get("sftpRemoteRelativePath").toString();
+        }
+
+        if (args.containsKey("sftpUsername")) {
+            sftpUsername = args.get("sftpUsername").toString();
+        }
+
+        if (sftpUsername == null) {
+            throw new IllegalArgumentException("Username is required to access an SFTP server");
+        }
+
+        // This key is used for logs and for getting the password from 3rd party credentials.
+        sftpUrlKey = "sftp://" + sftpUsername + "@" + sftpHostname + ":" + sftpPort;
```

Comment:
I really think the port is not necessary for the credentials key. You can create a different variable to store the url